lib/sinon/behavior.js

```diff
@@ -156,7 +158,12 @@ var proto = {
             return (this.promiseLibrary || Promise).reject(this.returnValue);
         } else if (this.callsThrough) {
             return this.stub.wrappedMethod.apply(context, args);
+        } else if (this.returnValueDefined) {
```

Comment:
Where is `returnValueDefined` being set? I don't see it anywhere else.

Reply:
Its set in `default-behaviour.js` for returns, and resolves. Not sure it was the perfect answer. Original thought of `typeof this.returnValue !== "undefined"`. Open to suggestions

Reply:
I have gone back and used `typeof this.returnValue !== "undefined"` as methods such as returnsThis don't set `returnValueDefined`